### ISSUE TITLE
Propagate `unused_variable` to variables that would otherwise be unused

### ISSUE THREAD

**Issue by Kampfkarren** [Kampfkarren/selene#357]
```lua
local ReplicatedStorage = game:GetService("ReplicatedStorage")

local t = require(ReplicatedStorage.Packages.t)
```

...currently reports that t is unused, but not ReplicatedStorage. That's not incorrect, ReplicatedStorage *is* used after all, but once you remove the t usage, then ReplicatedStorage becomes unused.

selene should report both as unused, but carry a message in the ReplicatedStorage one as to why it's being marked as unused.

I swear I've seen this propagation somewhere before, but have no idea where...it doesn't appear to be Rust or ESLint

**Comment by Yumacide** [comment]
I don't think this example should propagate the lint. A function call like `require` uses the variable, and assigning it to an unused variable shouldn't change that. Something a bit more conservative like linting `unused2 = unused1 + foo` or `unused2 = unused1` is more reasonable.
